cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -232,19 +211,52 @@ protected TimeSeries getTimeseries(String page, int pageSize, String names, Stri
 							.lessOrEqual(CWMS_UTIL_PACKAGE.call_TO_TIMESTAMP__2(DSL.val(endTime.toInstant().toEpochMilli())))
 					);
 
-			if(pageSize > 0)
-				query.limit(DSL.val(pageSize + 1));
+				if(pageSize > 0)
+					query.limit(DSL.val(pageSize + 1));
 
-			logger.finest( () -> query.getSQL(ParamType.INLINED));
+				logger.finest(() -> query.getSQL(ParamType.INLINED));
 
-			query.fetchInto(tsRecord -> timeseries.addValue(
-					tsRecord.getValue("DATE_TIME", Timestamp.class),
-					tsRecord.getValue("VALUE", Double.class),
-					tsRecord.getValue("QUALITY_CODE", Integer.class)
-			)
-			);
+				query.fetchInto(tsRecord -> timeseries.addValue(
+								tsRecord.getValue("DATE_TIME", Timestamp.class),
+								tsRecord.getValue("VALUE", Double.class),
+								tsRecord.getValue("QUALITY_CODE", Integer.class)
+						)
+				);
+				retval = timeseries;
+			}
+		} catch(org.jooq.exception.DataAccessException e){
+			if(isNotFound(e.getCause())){
+				throw new NotFoundException(e.getCause());
+			}
+			throw e;
 		}
-		return timeseries;
+		return retval;
+	}
+
+
+	private boolean isNotFound(Throwable cause)
+	{
+		boolean retval = false;
+		if(cause instanceof SQLException){
+			SQLException sqlException = (SQLException) cause;
+
+			// There may be better ways to figure out if the exception was b/c Not found.
+
+			// When we type in a garbage tsId the cause is a SQLException.  Its cause is an OracleDatabaseException.
+			String message = sqlException.getMessage();
+
+			// Note: The message I saw looked like this:
+			//ORA-20001: TS_ID_NOT_FOUND: The timeseries identifier "%1" was not found for office "%2"
```

Comment:
The 20001 is the error code you want to check for. That's specifically "no timeseries with that name found"